### ISSUE TITLE
Representation of function pointer types in the reflection stack

### ISSUE THREAD

**Issue by MichalStrehovsky** [dotnet/coreclr#20668]
dotnet/csharplang#1951 discusses adding function pointer types to the C# language. The CLR has supported these for a while, but there are areas (such as the reflection stack), where function pointer types are weird/unhandled.

While this will just be a corner area of the C# language, we should probably at least track this:

- [ ] Can we fix `ldtoken method explicit instance int32 *(int32)` not to return the token of `System.IntPtr`?
- [ ] Can we use `System.Reflection.Pointer` to reflection invoke methods that take/return function pointers?
- [ ] Do we need new APIs to inspect/reflection emit function pointer types?

**Comment by danmosemsft** [comment]
@MichalStrehovsky marked future.

**Comment by mjsabby** [comment]
Do you know off the of your head how complicated it is to do the first one?

**Comment by MichalStrehovsky** [comment]
> Do you know off the top of your head how complicated it is to do the first one?

No, I didn't look into it beyond trying that single line and being happy that I didn't have to add any extra work to .NET Native/CoreRT backlog at that time.
